demos/using-dev-build/README.md

```diff
@@ -1,16 +1,63 @@
+
+- `internet_identity`: The development version of Internet Identity.
+- `webapp`: A tiny webapp that calls out to the `internet_identity` canister for identity (anchor) creation and authentication, and that then calls the `whoami` canister (see below) to show that the identity is valid. You'll find the source of the webapp in [`index.html`](./index.html) and [`index.js`](./index.js).
+- `whoami`: A simple canister that checks that calls are authenticated, and that returns the "principal of the caller". The implementation is terribly simple:
+  ```motoko
+  actor {
+      public query ({caller}) func whoami() : async Principal {
+          return caller;
+      };
+  };
+  ```
+  On the IC, principals are basically the "identity" of someone performing a request or "call" (hence "caller"), and every call must be authenticated — i.e., must have a valid principal. When using Internet Identity you are using [self-authenticating principals](https://smartcontracts.org/docs/interface-spec/index.html#principal), which is a very fancy way of saying that you have a private key on your laptop (hidden behind TouchID, Windows Hello, etc) that your browser uses to sign and prove that you are indeed the person issuing the calls to the IC. If the IC actually lets the call (request) through to the `whoami` canister, it means that everything checked out, and the `whoami` canister just responds with the information the IC adds to requests, namely your identity (principal).
+
+Let's now use those canisters. Don't care about details? Skip to the [helpers](#helpers).
+
+In order to talk to those canisters (for instance to view the webapp in your browser) you need to figure the ID of each canister and then use an URL of the form `https://localhost:8000/?canisterId=<canister ID>` (where `8000` is the port used by `dfx` to proxy calls to the replica; that port is usually specified in the `dfx.json`). You can find the canister IDs in the output of the `dfx command`, or by checking `dfx`'s "internal" (read: non-documented) state:
+
 ```
+~/internet-identity/demos/using-dev-build$ cat .dfx/local/canister_ids.json {
+  "__Candid_UI": {
+    "local": "r7inp-6aaaa-aaaaa-aaabq-cai"
+  },
+  "internet_identity": {
+    "local": "rwlgt-iiaaa-aaaaa-aaaaa-cai"
+  },
+  "webapp": {
+    "local": "rrkah-fqaaa-aaaaa-aaaaq-cai"
+  },
+  "whoami": {
+    "local": "ryjl3-tyaaa-aaaaa-aaaba-cai"
+  }
+```
+
+You might get different canister IDs (and that's totally fine). If the `webapp` canister ID is `rrkah-fqaaa-aaaaa-aaaaq-cai`, you should be able to point your browser to [`http://localhost:8000/?canisterId=rrkah-fqaaa-aaaaa-aaaaq-cai`](http://localhost:8000/?canisterId=rrkah-fqaaa-aaaaa-aaaaq-cai) to see the webapp. Hurray!
+
+![](./webapp.png)
+
+_If you actually use the webapp, make sure that the "Internet Identity URL" field points to `http://localhost:8000/?canisterId=<canister ID of the internet_identity canister>`._
+
+## Helpers
```

Comment:
It sounds too much to helper functions to me (maybe just me). How about "Helpful commands"

Reply:
Heh, that might be just you indeed. Will leave as is unless you feel really strongly about it!